workflow/kf-rnaseqv4-wf.cwl

```diff
@@ -0,0 +1,878 @@
+  STAR-Fusion_results: {type: 'File', outputSource: star_fusion_1-10-1/abridged_coding,
+    doc: "STAR fusion detection from chimeric reads"}
+  arriba_fusion_results: {type: 'File', outputSource: arriba_fusion_2-2-1/arriba_fusions,
+    doc: "Fusion output from Arriba"}
+  arriba_fusion_viz: {type: 'File', outputSource: arriba_draw_2-2-1/arriba_pdf, doc: "pdf\
+      \ output from Arriba"}
+  RSEM_isoform: {type: 'File', outputSource: rsem/isoform_out, doc: "RSEM isoform\
+      \ expression estimates"}
+  RSEM_gene: {type: 'File', outputSource: rsem/gene_out, doc: "RSEM gene expression\
+      \ estimates"}
+  RNASeQC_Metrics: {type: 'File', outputSource: rna_seqc/Metrics, doc: "Metrics on\
+      \ mapping, intronic, exonic rates, count information, etc"}
+  RNASeQC_counts: {type: 'File', outputSource: supplemental/RNASeQC_counts, doc: "Contains\
+      \ gene tpm, gene read, and exon counts"}
+  kallisto_Abundance: {type: 'File', outputSource: kallisto/abundance_out, doc: "Gene\
+      \ abundance output from STAR genomic bam file"}
+  annofuse_filtered_fusions_tsv: {type: 'File?', outputSource: annofuse/annofuse_filtered_fusions_tsv,
+    doc: "Filtered fusions called by annoFuse."}
+  rmats_filtered_alternative_3_prime_splice_sites_jc: {type: 'File', outputSource: rmats/filtered_alternative_3_prime_splice_sites_jc,
+    doc: "Alternative 3 prime splice sites JC.txt output from RMATs containing only\
+      \ those calls with 10 or more read counts of support"}
+  rmats_filtered_alternative_5_prime_splice_sites_jc: {type: 'File', outputSource: rmats/filtered_alternative_5_prime_splice_sites_jc,
+    doc: "Alternative 5 prime splice sites JC.txt output from RMATs containing only\
+      \ those calls with 10 or more read counts of support"}
+  rmats_filtered_mutually_exclusive_exons_jc: {type: 'File', outputSource: rmats/filtered_mutually_exclusive_exons_jc,
+    doc: "Mutually exclusive exons JC.txt output from RMATs containing only those\
+      \ calls with 10 or more read counts of support"}
+  rmats_filtered_retained_introns_jc: {type: 'File', outputSource: rmats/filtered_retained_introns_jc,
+    doc: "Retained introns JC.txt output from RMATs containing only those calls with\
+      \ 10 or more read counts of support"}
+  rmats_filtered_skipped_exons_jc: {type: 'File', outputSource: rmats/filtered_skipped_exons_jc,
+    doc: "Skipped exons JC.txt output from RMATs containing only those calls with\
+      \ 10 or more read counts of support"}
+
+steps:
+  bam2fastq:
+    # Skip if input is FASTQ already
+    run: ../tools/samtools_fastq.cwl
+    when: $(inputs.input_type != "FASTQ")
+    in:
+      input_reads_1: reads1
+      input_reads_2: reads2
+      SampleID: output_basename
+      cores: samtools_fastq_cores
+      input_type: input_type
+    out: [fq1, fq2]
+
+  cutadapt:
+    # Skip if no adapter given, get fastq from prev step if not null or wf input
+    run: ../tools/cutadapter.cwl
+    when: $(inputs.r1_adapter != null)
```

Comment:
again this tool has conditional behavior that is now redundant/pointless. is it worth keeping the old tool around?

Reply:
Hmmm, could refactor both as it's still functionally equivalent, right?

Reply:
Yeah, there's no problem with this functionally. But if someone ever goes poking around in the tool they're going to have to wade through all that defunct code. If we're making a big deal about having a new workflow I think we should clean up the tools while we're at it. The easiest time to make changes is right now.